tesseract_motion_planners/include/tesseract_motion_planners/simple/profile/simple_planner_fixed_size_assign_plan_profile.h

```diff
@@ -0,0 +1,76 @@
+/**
+ * @file simple_planner_default_plan_profile.h
+ * @brief
+ *
+ * @author Matthew Powelson
+ * @date July 23, 2020
+ * @version TODO
+ * @bug No known bugs
+ *
+ * @copyright Copyright (c) 2020, Southwest Research Institute
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef TESSERACT_MOTION_PLANNERS_SIMPLE_FIXED_SIZE_ASSIGN_PLAN_PROFILE_H
+#define TESSERACT_MOTION_PLANNERS_SIMPLE_FIXED_SIZE_ASSIGN_PLAN_PROFILE_H
+
+#include <tesseract_common/macros.h>
+TESSERACT_COMMON_IGNORE_WARNINGS_PUSH
+#include <trajopt_sco/modeling_utils.hpp>
+#include <Eigen/Geometry>
+TESSERACT_COMMON_IGNORE_WARNINGS_POP
+
+#include <tesseract_motion_planners/simple/profile/simple_planner_profile.h>
+#include <tesseract_motion_planners/simple/profile/simple_planner_utils.h>
+
+#ifdef SWIG
+%shared_ptr(tesseract_planning::SimplePlannerFixedSizeAssignPlanProfile)
+#endif  // SWIG
+
+namespace tesseract_planning
+{
+class SimplePlannerFixedSizeAssignPlanProfile : public SimplePlannerPlanProfile
+{
+public:
+  using Ptr = std::shared_ptr<SimplePlannerFixedSizeAssignPlanProfile>;
+  using ConstPtr = std::shared_ptr<const SimplePlannerFixedSizeAssignPlanProfile>;
+
+  /**
+   * @brief SimplePlannerFixedSizeAssignPlanProfile
+   * @param freespace_steps The number of steps to use for freespace instruction
+   * @param linear_steps The number of steps to use for linear instruction
+   */
+  SimplePlannerFixedSizeAssignPlanProfile(int freespace_steps = 10, int linear_steps = 10);
+
+  CompositeInstruction generate(const PlanInstruction& prev_instruction,
+                                const PlanInstruction& base_instruction,
+                                const PlannerRequest& request,
+                                const ManipulatorInfo& global_manip_info) const override;
+
+  int getFreespaceSteps();
+  void setFreespaceSteps(int freespace_steps);
+
+  int getLinearSteps();
+  void setLinearSteps(int linear_steps);
+
+protected:
+  int freespace_steps_;
+  int linear_steps_;
```

Comment:
Is it relevant to interpolate freespace or linear moves with a different number of states? I can't really think of a scenario where a single number of steps wouldn't suffice for 99% of our uses cases

Reply:
Currently we use different values for each. I would agree that they are close if a configuration of the robots are the same but if it requires a configuration flip then the freespace would need to be higher. One example would be to assign this for the transitions and this single profile would be configured to handle both freespace and linear types where for freespace you would want to set higher to allow for configuration changes.

Reply:
Would the user of this class actually know about a configuration change between the two input states ahead of time and be able to change this parameter? It seems like the LVS profile should used if you care about how far apart the interpolated waypoints should be, and that the fixed size profiles should be used when a user just needs to specify some arbitrary number of steps/states (in which case the distinction between freespace and linear moves doesn't really matter).

Reply:
> Would the user of this class actually know about a configuration change between the two input states ahead of time and be able to change this parameter?

Yes and no, they have information about the process and would have some idea if it may be possible. In the case where you think it would be possible you would set the freespace to larger. As you mention the LVS is the default because it uses more information to try and determine what the ideal number of steps would be.

A profile is intended to handle the supported instruction types (FREESPACE and LINEAR) which an instruction can only be one of these which is why you have the ability to configure them individually.